packages/braintree-integration/src/braintree-fastlane/braintree-fastlane-payment-strategy.ts

```diff
@@ -198,20 +220,38 @@ export default class BraintreeAcceleratedCheckoutPaymentStrategy implements Paym
         // Info: shipping can be unavailable for carts with digital items
         const shippingAddress = state.getShippingAddress();
 
-        const deviceSessionId = await this.braintreeAcceleratedCheckoutUtils.getDeviceSessionId();
+        const deviceSessionId = await this.braintreeFastlaneUtils.getDeviceSessionId();
 
         const { shouldSaveInstrument = false, shouldSetAsDefaultInstrument = false } =
             isHostedInstrumentLike(paymentData) ? paymentData : {};
 
-        const braintreeConnectCreditCardComponent = this.getBraintreeCardComponentOrThrow();
+        const braintreeCreditCardComponent = this.getBraintreeCardComponentOrThrow();
 
         const paypalBillingAddress = this.mapToPayPalAddress(billingAddress);
         const paypalShippingAddress = shippingAddress && this.mapToPayPalAddress(shippingAddress);
 
-        const { nonce } = await braintreeConnectCreditCardComponent.tokenize({
-            billingAddress: paypalBillingAddress,
-            ...(paypalShippingAddress && { shippingAddress: paypalShippingAddress }),
-        });
+        let token;
+
+        if (
+            this.isFastlaneEnabled &&
+            isBraintreeFastlaneCardComponent(braintreeCreditCardComponent)
+        ) {
+            const { id } = await braintreeCreditCardComponent.getPaymentToken({
+                billingAddress: paypalBillingAddress,
+            });
+
+            token = id;
+        } else if (
+            !this.isFastlaneEnabled &&
```

Comment:
I don't think its necessary to have `!this.isFastlaneEnabled` here

Reply:
removed